CMakeLists.txt

```diff
@@ -133,6 +133,43 @@ endif()
 find_package(Threads REQUIRED)
 find_package(ZLIB REQUIRED)
 
+# try to find an installed boost
+set(boost_VERSION "1.71")
+find_package(Boost ${boost_VERSION} QUIET)
+if (NOT Boost_Found)
+  # bail if there's no conan installed
+  message(STATUS "No compatible boost version detected, using conan...")
+  find_program(conan_FOUND conan)
+  if (NOT conan_FOUND)
+    message(FATAL_ERROR "conan needs to be installed for boost, see docs.conan.io/en/latest/installation.html.")
+  endif()
+
+  if(NOT EXISTS "${CMAKE_BINARY_DIR}/conan.cmake")
+    message(STATUS "Valhalla: Downloading conan.cmake from https://github.com/conan-io/cmake-conan")
+    file(DOWNLOAD "https://raw.githubusercontent.com/conan-io/cmake-conan/0.17.0/conan.cmake"
+                  "${CMAKE_BINARY_DIR}/conan.cmake"
+                  EXPECTED_HASH SHA256=3bef79da16c2e031dc429e1dac87a08b9226418b300ce004cc125a82687baeef
+                  TLS_VERIFY ON)
+  endif()
+
```

Comment:
can we just pull this down and keep it then you dont need to do the module path stuff again below since we do it right at the beginning